src/box/alter.cc

```diff
@@ -4078,10 +4079,42 @@ on_commit_replicaset_name(struct trigger *trigger, void * /* event */)
 	return 0;
 }
 
+static int
+on_commit_dd_version_start_filtering(va_list /* ap */)
+{
+	txn_limbo_filter_enable(&txn_limbo);
+	return 0;
+}
+
+static int
+on_commit_dd_version_stop_filtering(va_list /* ap */)
+{
+	txn_limbo_filter_disable(&txn_limbo);
+	return 0;
+}
+
 static int
 on_commit_dd_version(struct trigger *trigger, void * /* event */)
 {
+	uint32_t old_version_id = dd_version_id;
 	dd_version_id = (uint32_t)(uintptr_t)trigger->data;
+	if (recovery_state != FINISHED_RECOVERY)
+		return 0;
+	if (dd_version_id > version_id(2, 10, 1) &&
+	    old_version_id <= version_id(2, 10, 1)) {
+		struct fiber *fiber;
+		fiber = fiber_new_system("synchro_queue_filter_enabler",
```

Comment:
Do I understand correctly, that we start a fiber here, because `synchro_queue_filter_enabler` might yield under the hood on obtaining `limbo->promote_latch`, and the trigger shouldn't yield?

Reply:
Yes, exactly.